src/.internal/baseEach.ts

```diff
@@ -13,7 +13,7 @@ function baseEach(collection, iteratee) {
   if (collection == null) {
     return collection
   }
-  if (!isArrayLike(collection)) {
+  if (!isArrayLike(collection) || collection.constructor?.name === 'Object') {
```

Comment:
To me, this looks like a breaking change.

The current implementation works well for JS classes like this:

```
class Rectangle {
  constructor(height, width) {
    this.height = height;
    this.width = width;
  }
}

_.reduce(new Rectangle(100, 100), function(result, value, key) {
  result[key] = value;
  return result;
}, {});
```

However, this change is now checking the constructor name, and `collection.constructor?.name === 'Object'` would return `false` to `Rectangle`, so it would probably break there.